src/components/generics/MainMenuContribution.js

```diff
@@ -128,7 +129,52 @@ class MainMenuContribution extends Component {
     _historyPush(modulesManager, history, route);
   }
 
-  appBarMenu = () => {
+  fetchSubmenuConfig(modulesManager, allEntries) {
+    const menuConfig = modulesManager.getConf("fe-core", "menus", []);
+    const isMenuConfigEmpty = !menuConfig || menuConfig.length === 0;
+    const submenuMapping = {};
+    const copyOfEntries = this.props.entries;
```

Comment:
If `this.props.entries` can be passed into this function as an argument then this `fetchSubmenuConfig` function can be pulled out of scope of class `MainMenuContribution` as a pure function? Alternatively, consider wrapping `fetchSubmenuConfig` in `React.useCallback` with explicit dependencies to avoid unnecessary re-rendering?

Reply:
```
render() {
    const { menuVariant, modulesManager } = this.props;
    const allEntries = modulesManager.getMenuEntries();
    const updatedEntries = this.fetchSubmenuConfig(modulesManager, allEntries);
    if (menuVariant === "AppBar") {
      return this.appBarMenu(updatedEntries);
    } else {
      return this.drawerMenu(updatedEntries);
    }
  }
``` 
On line 260 there is a call of the 'fetchSubmenuConfig'. I don't think the first approach could work. We can think about 2nd option you proposed with useCallback. I can check this today or tomorrow and try with that approach.

Reply:
Ahh, sorry, 2nd approach won't pass - here we have class component, not functional one. Using hook 'React.useCallback' in that case would be impossible without rewriting to functional component.

Reply:
Let's go back into first approach.

Reply:
First approach - I applied https://github.com/openimis/openimis-fe-core_js/pull/262/commits/d5be58cd1581c840f416b5adb726b2efe4d96a45 here the fix (which will be merged into this PR)